SYCL/ESIMD/api/functional/type_traits.hpp

```diff
@@ -0,0 +1,38 @@
+//===-- type_traits.hpp - Define functions for iterating with datatypes. --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides function for iterating with data types.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <type_traits>
+
+namespace esimd_test {
+namespace api {
+namespace functional {
+namespace type_traits {
+
+template <typename T> constexpr bool is_sycl_type() {
+  return std::is_same_v<T, sycl::half>;
+}
+
+template <typename T> struct is_floating_point {
+  static constexpr bool value{std::is_floating_point_v<T> ||
+                              std::is_same_v<T, sycl::half>};
+};
```

Comment:
```suggestion
template <typename T>
using is_sycl_floating_point =
  std::bool_constant<std::is_floating_point_v<T> ||
                                  std::is_same_v<T, sycl::half>>;

```
What do you think? It's available in C++17
1) At the closer look we have name clash wit std::is_floating_point here.
2) By using type alias we avoid creating any new type

But we will need to change the `is_sycl_type()` naming appropriately in case of such change. May be `is_sycl_specific<T>()`?

Reply:
this comment was applied in this commit: ef6509579513a2452045f0c2cf2c0599a6c3b96b

Reply:
So you've preferred to retain the `is_sycl_type()` naming?
Actually.. Do we even use it? May be we should just remove the `is_sycl_type()` function?

Reply:
> May be we should just remove the is_sycl_type() function?

Yes, this function was removed in this commit: this function was removed in this commit: 465eb5fe27f50118853554671143841f92d49407